src/python/pants/backend/python/goals/run_pex_binary_integration_test.py

```diff
@@ -124,3 +125,36 @@ def test_no_strip_pex_env_issues_12057() -> None:
         ]
         result = run_pants(args)
         assert result.exit_code == 42, result.stderr
+
+
+def test_no_leak_pex_root_issues_12055() -> None:
+    read_config_result = run_pants(["help-all"])
+    read_config_result.assert_success()
+    config_data = json.loads(read_config_result.stdout)
+    global_advanced_options = {
+        option["config_key"]: [
+            ranked_value["value"] for ranked_value in option["value_history"]["ranked_values"]
+        ][-1]
+        for option in config_data["scope_to_help_info"][""]["advanced"]
+    }
+    named_caches_dir = global_advanced_options["named_caches_dir"]
```

Comment:
This is a bit insane but I don't know of a better way right now.

Reply:
Nor me.